src/commands/viewProperties.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IActionContext, openReadOnlyJson } from '@microsoft/vscode-azext-utils';
+import { rootFilter } from '../constants';
+import { ext } from '../extensionVariables';
+import { azureResourceRegExp, IAzureResourceTreeItem } from '../tree/IAzureResourceTreeItem';
+import { localize } from '../utils/localize';
+import { nonNullProp } from '../utils/nonNull';
+
+export async function viewProperties(context: IActionContext, node?: IAzureResourceTreeItem): Promise<void> {
+    if (!node) {
+        node = await ext.rgApi.pickAppResource<IAzureResourceTreeItem>(context, {
+            filter: rootFilter,
+            expectedChildContextValue: azureResourceRegExp
+        });
+    }
+
+    if (!node.data) {
+        if (node.getDataImpl) {
+            await node.getDataImpl();
+        } else {
+            throw new Error(localize('No data exists on resource "{0}"', node.label));
```

Comment:
`localize` needs an ID for each string;

```suggestion
            throw new Error(localize('viewProperties.noData', 'No data exists on resource "{0}"', node.label));
```

As written, it would think the "No data exists..." is the ID and `node.label` is the string to show